bin/varnishd/cache/cache_fetch.c

```diff
@@ -1043,9 +1043,8 @@ vbf_stp_fail(struct worker *wrk, struct busyobj *bo)
 	CHECK_OBJ_NOTNULL(oc, OBJCORE_MAGIC);
 
 	assert(oc->boc->state < BOS_FINISHED);
-	HSH_Fail(oc);
-	if (!(oc->flags & OC_F_BUSY))
```

Comment:
Um this does not seem to make sense to me. The `EXP_Insert()` does not accept `OC_F_BUSY` objects, so why would we `HSH_Kill()` it?

Reply:
Because with this change `HSH_Fail()` will now make sure that the objcore is unbusied, see commit message:

> If the fetch task failed before reaching HSH_Unbusy() we make sure to
signal the waiting list immediately and not wait for the final objcore
reference to be dropped.

The important thing here is that a fetch task always clears the busy flag, for all outcomes. In this case, sooner than later, where later means when the last objcore ref is dropped (aka way too late).

Reply:
I will also mark this one as resolved. The meaning of the busy flag is covered by the previous-previous comment thread.